### DISCUSSION TITLE
how to fix an encoding issue with Chinese characters when piping in a Windows console?

### DISCUSSION THREAD

**Discussion by cesaryuan** [BurntSushi/ripgrep#2511] [Q&A] [answered]
#### What version of ripgrep are you using?

ripgrep 13.0.0 (rev af6b6c543b)
-SIMD -AVX (compiled)
+SIMD +AVX (runtime)

#### How did you install ripgrep?

Scoop

#### What operating system are you using ripgrep on?

Windows 10 19045.2846
PowerShell 7.3.4

#### Describe your bug.

There seems to be an encoding issue when piping the output of a ripgrep search. The chinese characters `测试` (and probably others, those are just the ones I've tested with) are converted to wrong characters(`娴嬭瘯`) when being piped to anything.

Passing stdout to a file with > is also with this problem.

This is what I used to test.

```
PS D:\0-Download> echo "测试Chinese Test" | rg -o "测试"
测试
PS D:\0-Download> echo "测试Chinese Test" | rg -o "测试" | echo
娴嬭瘯
PS D:\0-Download> echo "测试Chinese Test" | rg -o "测试" > test.txt
PS D:\0-Download> cat .\test.txt
娴嬭瘯
```

![image](https://github.com/BurntSushi/ripgrep/assets/35998162/eaa37a24-9800-4862-9f15-5189cab8900e)

#### What are the steps to reproduce the behavior?

```
PS D:\0-Download> echo "测试Chinese Test" | rg -o "测试"
PS D:\0-Download> echo "测试Chinese Test" | rg -o "测试" | echo
PS D:\0-Download> echo "测试Chinese Test" | rg -o "测试" > test.txt
PS D:\0-Download> cat .\test.txt
```

#### What is the actual behavior?

```
PS D:\0-Download> echo "测试Chinese Test" | rg -o --debug "测试" | echo
DEBUG|grep_regex::literal|crates\regex\src\literal.rs:58: literal prefixes detected: Literals { lits: [Complete(测试)], limit_size: 250, limit_class: 10 }
DEBUG|grep_cli::decompress|crates\cli\src\decompress.rs:482: gzip: could not find executable in PATH
DEBUG|grep_cli::decompress|crates\cli\src\decompress.rs:482: gzip: could not find executable in PATH
DEBUG|grep_cli::decompress|crates\cli\src\decompress.rs:482: bzip2: could not find executable in PATH
DEBUG|grep_cli::decompress|crates\cli\src\decompress.rs:482: bzip2: could not find executable in PATH
DEBUG|grep_cli::decompress|crates\cli\src\decompress.rs:482: lz4: could not find executable in PATH
DEBUG|grep_cli::decompress|crates\cli\src\decompress.rs:482: brotli: could not find executable in PATH
DEBUG|grep_cli::decompress|crates\cli\src\decompress.rs:482: uncompress: could not find executable in PATH
DEBUG|globset|crates\globset\src\lib.rs:421: built glob set; 0 literals, 0 basenames, 5 extensions, 0 prefixes, 0 suffixes, 0 required extensions, 0 regexes
娴嬭瘯
```

#### What is the expected behavior?

```
PS D:\0-Download> echo "测试Chinese Test" | rg -o "测试" | echo
测试
```

**Comment by cesaryuan** [comment, marked as answer]
https://github.com/BurntSushi/ripgrep/blob/041544853c86dde91c49983e5ddd0aa799bd2831/FAQ.md?plain=1#L810-L814

Tried this. Now works!

**Comment by BurntSushi** [comment]
Nice, glad you got it figured out and thank you for posting the solution!
